### DISCUSSION TITLE
Cannot register type for structures with Boxed values in bevy_reflect

### DISCUSSION THREAD

**Discussion by Escapingbug** [bevyengine/bevy#9886] [Q&A]
When dering `Reflect` for type like:

```rust
#[derive(Reflect)]
struct MyTest(Box<String>);
```

The deriving part seems to be Ok, but when later register the type by:

```rust
let mut reg = TypeRegistry::default();
reg.register::<MyTest>();
```

It reported:

```
the trait `FromReflect` is not implemented for `Box<std::string::String>`
```

Is this expected result? Is there any workaround to bypass this limitation?

**Comment by MrGVSV** [comment]
This is because `Box<T>` does not implement any reflection trait, including `FromReflect`. There are some related issues and PRs:
- https://github.com/bevyengine/bevy/issues/3392
- https://github.com/bevyengine/bevy/pull/3400
- https://github.com/bevyengine/bevy/pull/6098

I'm a bit surprised you were able to derive it. Did you not get a compile error from that?

**Reply by Escapingbug** [reply]
Plain derive does not give any error.
The error only comes when I'm trying to register that type.

So if I'm right about this, by implementing #6098, I will be able to deal with this issue?
I'm currently on a project which must have this feature. If it was so, I think I may help finish that.

**Reply by MrGVSV** [reply]
> Plain derive does not give any error.
> The error only comes when I'm trying to register that type.

I just tested the same code and the derive does fail for me. What version are you on?

> So if I'm right about this, by implementing #6098, I will be able to deal with this issue?
> I'm currently on a project which must have this feature. If it was so, I think I may help finish that.

Yeah I think so. If you want to adopt the PR, just be sure to reach out to the author on it and see if they'd be okay with that. I would also link back to this discussion and share your use case as well.

**Reply by Escapingbug** [reply]
Should be 0.11.2 bevy reflect.
